src/com/goide/inspections/GoStructTagInspection.java

```diff
@@ -0,0 +1,107 @@
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.psi.GoFieldDeclaration;
+import com.goide.psi.GoStructType;
+import com.goide.psi.GoTag;
+import com.goide.psi.GoVisitor;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.ProblemsHolder;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Implements <a href="https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/1983"/>, an
+ * inspector that warns if a go StructTag is not well-formed according to Go language conventions.
+ */
+public class GoStructTagInspection extends GoInspectionBase {
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder,
+                                     @SuppressWarnings({"UnusedParameters", "For future"}) @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitStructType(@NotNull GoStructType o) {
+        for (GoFieldDeclaration field : o.getFieldDeclarationList()) {
+          if (field.getTag() == null) {
+            continue;
+          }
+          if (!isValidTag(field.getTag())) {
+            holder.registerProblem(field.getTag(), "Bad syntax for struct tag value");
+          }
+        }
+      }
+    };
+  }
+
+  // Implementation based on validateStructTag from the go vet tool:
+  // https://github.com/golang/tools/blob/master/cmd/vet/structtag.go.
+  private boolean isValidTag(GoTag tag) {
+    StringBuilder tagText = new StringBuilder(tag.getText().substring(1, tag.getTextLength() - 1));
+    while (tagText.length() > 0) {
+      int i;
+      i = 0;
+
+      // Skip leading spaces.
+      while (i < tagText.length() && tagText.charAt(i) == ' ') {
+        i++;
+      }
+
+      tagText.delete(0, i);
+      if (tagText.length() == 0) {
+        return true;
+      }
+
+      // Scan to colon. A space, a quote or a control character is a syntax error.
+      // Strictly speaking, control chars include the range [0x7f, 0x9f], not just
+      // [0x00, 0x1f], but in practice, we ignore the multi-byte control characters
+      // as it is simpler to inspect the tag's bytes than the tag's runes.
+      i = 0;
+      while (i < tagText.length() &&
+             tagText.charAt(i) > ' ' &&
+             tagText.charAt(i) != ':' &&
+             tagText.charAt(i) != '"' &&
+             tagText.charAt(i) != 0x7f) {
+        i++;
+      }
+      if (i == 0 || (i + 1) > tagText.length() || tagText.charAt(i) != ':') {
+        return false;
+      }
+      tagText.delete(0, i + 1);
+
+      // Scan quoted string to find value.
+      i = 1;
+      while (i < tagText.length() && tagText.charAt(i) != '"') {
+        if (tagText.charAt(i) == '\\') {
+          i++;
+        }
+        i++;
+      }
+
+      if (i >= tagText.length()) {
+        return false;
+      }
+
+      // TODO(sjamesr): check strconv.Unquote(quotedValue)
```

Comment:
As I understand it, that method isn't equivalent to strconv.Unquote because it won't resolve escape sequences within the string. Also strconv.Unquote will return errors if it finds mismatched quotes, bad escape sequences, etc.

What should I do? Right now I'm tempted just to do:
    quotedValue.size() >= 2 && quotedValue.charAt(0) == quotedValue.charAt(quotedValue.size() - 1) && isQuote(quotedValue.charAt(0))

This will give us a pretty good start.

Is `GoStringLiteralEscaper#decode` appropriate? We could decode what's inside the quotes using that method, if it returns true, assume the quoted value is valid.